### PR TITLE
FIX(Graphql): Add support for input with default values

### PR DESCRIPTION
**Pull request by eMerzh**
Hello here 👋 

We have a few inputs with default required fields that have default values like here :

```graphql
input StoryPinInput {
  id: ID!
  isGlobal: Boolean! = false
}
```

but the TS generated for this look like 
```ts
export type StoryPinInput = {
  id: string;
  isGlobal: boolean;
};
```
which is not ideal ... 

so i marked the field as optional if it has a default :)

### CODE REVIEW

**Comment by eMerzh** [comment]
@captbaritone sorry for the ping 🥹 .... is there a way to bump those PRs before they go down the black hole of 2nd page ? 

if there's something we can do to help ? 🤷

**Comment by captbaritone** [comment]
Sorry for the delayed response. Can you add a fixture test here? https://github.com/facebook/relay/tree/main/compiler/crates/relay-typegen/tests/generate_typescript/fixtures I think the test schema has an example you can trigger here: https://github.com/facebook/relay/blob/54586eccb033cbdd0896bb1d5d53727a523e3032/packages/relay-test-utils-internal/testschema.graphql#L677

Maybe you could also ensure there'a fixture test that validates what we do for Flow in this case?

Info on updating fixture tests can be found here: https://github.com/facebook/relay/blob/main/.github/CONTRIBUTING.md#pull-requests

Defaults in GraphQL are confusing, but I believe we should should ensure this generates `{ isGlobal?: boolean }` and not `{ isGlobal: boolean | null | undefined }` since I believe with required arguments with defaults you are permitted to omit the argument but not pass an explicit null.

**Comment by eMerzh** [comment]
awesome! ❤️  .... i'll take a look at that

**Comment by eMerzh** [comment]
here... i hope it's better now :) 

tried to not add input but i prefer avoid touching too much other tests to add this

File: compiler/crates/relay-typegen/tests/generate_typescript/fixtures/default-input.graphql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+mutation feedbackUnLikeMutation($input: FeedbackUnLikeInput) {
+  feedbackUnLike(input: $input) {
+    feedback {
+      id
+    }
+  }
+}
```
Comment (captbaritone) [inline compiler/crates/relay-typegen/tests/generate_typescript/fixtures/default-input.graphql:7]: Would you mind adding this same fixture to the Flow tests?

https://github.com/facebook/relay/tree/main/compiler/crates/relay-typegen/tests/generate_flow/fixtures

**Review by captbaritone — CHANGES_REQUESTED** [review]

File: compiler/crates/relay-typegen/tests/generate_typescript/fixtures/default-input.expected [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  feedbackUnLike(input: $input) {
+    feedback {
+      id
+    }
+  }
+}
+==================================== OUTPUT ===================================
+export type FeedbackUnLikeInput = {
+  feedbackId?: string | null | undefined;
+  silent?: boolean;
```
Comment (captbaritone) [inline compiler/crates/relay-typegen/tests/generate_typescript/fixtures/default-input.expected:12]: nice!

File: compiler/crates/relay-typegen/tests/generate_typescript/fixtures/default-input.graphql (hunk 1 above)
Comment (eMerzh) [reply compiler/crates/relay-typegen/tests/generate_typescript/fixtures/default-input.graphql:7]: arg, my bad... here they are :)

**Comment by eMerzh** [comment]
awesome ❤️  thanks for the review

**Pull request closed without merging**